### PR TITLE
LoopyPyOpenCLTarget: pass through loopy.PyOpenCLTarget

### PR DESCRIPTION
**Pull request by matthiasdiener**
Needs:
- ~~https://github.com/inducer/loopy/pull/642~~

### CODE REVIEW

**Comment by inducer** [comment]
Not sure I love this. In general, we could just set the target on the generated loopy translation unit after it has been generated. This would make pytato's logic entirely independent of the details of loopy target selection, which seems nicer. @kaushikcfd, would you agree?

**Comment by kaushikcfd** [comment]
> This would make pytato's logic entirely independent of the details of loopy target selection, which seems nice

I agree, I think the principled way to do this is not leaking any details about the loopy kernel execution to `generate_loopy`.

File: pytato/target/loopy/__init__.py [hunk 1]
```diff
@@ -85,17 +85,25 @@ class LoopyPyOpenCLTarget(LoopyTarget):
...
         :class:`loopy.PyOpenCLTarget`, or *None*.
+
+    .. attribute:: limit_arg_size_nbytes
+
+        An optional int that indicates the size limit of kernel arguments
+        and gets passed to :class:`loopy.PyOpenCLTarget`.
     """
 
-    def __init__(self, device: Optional["pyopencl.Device"] = None):
+    def __init__(self, device: Optional["pyopencl.Device"] = None,
```
Comment (matthiasdiener) [inline pytato/target/loopy/__init__.py:95]: Create a new constructor that accepts an `lp.PyOpenCLTarget()`

Comment (matthiasdiener) [reply pytato/target/loopy/__init__.py:95]: Is https://github.com/inducer/pytato/pull/359/commits/2942fbfd452bfbdad5b9fbc5e4b637cdb896e9ad what you had in mind @inducer?

**Comment by matthiasdiener** [comment]
Testing this together with https://github.com/inducer/arraycontext/pull/189 appears to work as expected, and avoids the test failures seen in https://github.com/illinois-ceesd/mirgecom/pull/746

File: pytato/target/loopy/__init__.py (hunk 1 above)
Comment (inducer) [reply pytato/target/loopy/__init__.py:95]: Sorry to be sending you around in circles, but I just realized we don't even need any pytato changes. We can simply subclass `LoopyPyOpenCLTarget` in the array context and override `get_loopy_target` there.

Comment (matthiasdiener) [reply pytato/target/loopy/__init__.py:95]: What do you think of https://github.com/inducer/arraycontext/pull/189/commits/2768feea8d7e4846d3c6df731d46238c8f776375 ?

**Comment by matthiasdiener** [comment]
I think this PR can be closed right?

**Comment by inducer** [comment]
Yep.

**Pull request closed without merging**